core/math/random_pcg.cpp

```diff
@@ -39,7 +39,7 @@ RandomPCG::RandomPCG(uint64_t p_seed, uint64_t p_inc) :
 }
 
 void RandomPCG::randomize() {
-	seed(OS::get_singleton()->get_ticks_usec() * pcg.state + PCG_DEFAULT_INC_64);
+	seed(OS::get_singleton()->get_unix_time() * pcg.state + PCG_DEFAULT_INC_64);
```

Comment:
Unix time is in seconds, I wonder if it's too coarse. You might still be likely to get the same seed if starting several processes in parallel. Maybe we should indeed sparkle some `get_ticks_usec()` here additionally to `get_unix_time()`? Not sure if that's worth the trouble.

CC @Faless

Reply:
I meant if you start your game multiple times in different OS processes (e.g. to test multiplayer locally).

Reply:
With multiplayer, that could be a feature tbh, because usually scenes will be synchronized across instances/processes. If you don't synchronize scenes manually, then the same seed should produce the same level geometry etc.

Reply:
The only downside of this is that calling `randomize()` 2 times consecutively would likely return the same number, which is confusing:
```gdscript
randomize()
var x = randi()
randomize()
var y = randi()
assert(x == y) # Wil likely be true.
```

Reply:
Other sources that could be used to get a random-ish number:

- Process ID
- Thread ID

Reply:
> The only downside of this is that calling `randomize()` 2 times consecutively would likely return the same number, which is confusing:

This makes me think that `randomize()` should actually be `initialize()`, which can optionally take a user seed, otherwise it will use time-based seed by default. I don't think that `randomize()` is intended to be used more than once per application lifetime (I recall this was mentioned somewhere in docs as well).

Reply:
Would something like `get_unix_time() + get_ticks_usec()` be better?